bookkeeper-server/src/test/java/org/apache/bookkeeper/test/PortManager.java

```diff
@@ -32,19 +30,18 @@
     private static int nextPort = getBasePort();
 
     public static synchronized int nextFreePort() {
+        int exceptionCount = 0;
         while (true) {
-            ServerSocket ss = null;
-            try {
-                int port = nextPort++;
-                ss = new ServerSocket(port);
-                ss.setReuseAddress(true);
+            int port = nextPort++;
+            try (ServerSocket ss = new ServerSocket(port)) {
```

Comment:
@eolivelli

1) this change doesn't change the logic here.
2) this class is a general test class. so whether a service can use port 0, not depends on this class, but depends on the service logic. some service (e.g. bookie) at some test cases can use port 0, some can't. whether a service can or not is a question of the service not the util.

Reply:
Sorry, my idea was:
s = new ServerSocket(0).
return s.getLocalPort()

Reply:
Sorry, my idea was:
s = new ServerSocket(0).
return s.getLocalPort()

Reply:
@eolivelli even you change to use port 0, you still the logic here. the change is to wait for releasing port, the change is about how to allocate the ports. if you feel we should switch the port election to other approach, please file a separate issue. it would be easier to keep one issue focus its own problem. "using port 0 or not" is different from "waiting for os to release".